NitroxClient/GameLogic/Vehicles.cs

```diff
@@ -106,5 +120,170 @@ private void OnVehiclePrefabLoaded(GameObject prefab, string guid, Vector3 spawn
+            packetSender.Send(vehicleAdd);
+        }
+
+        public void Add(VehicleModel vehicleModel) //Add Vehicle From Save
+        {
+            CreateVehicleAt(vehicleModel.TechType, vehicleModel.Guid, vehicleModel.Position, vehicleModel.Rotation);
+        }
+
+        public void Remove(Vehicle vehicle) //Remove Vehicle From Instance Event OnKill
+        {
+            using (packetSender.Suppress<VehicleOnPilotMode>())
+            {
+                string guid = GuidHelper.GetGuid(vehicle.gameObject);
+                LocalPlayer localPlayer = NitroxServiceLocator.LocateService<LocalPlayer>();
+                VehicleRemoveEntry vehicleremove = new VehicleRemoveEntry(guid, localPlayer.PlayerName, vehicle.GetPilotingMode());
+                packetSender.Send(vehicleremove);
+
+                GuidHelper.SetNewGuid(vehicle.gameObject, string.Empty); //Clean Local (Detach Player From Vehicle Call OnPilotMode Event if Guid is Empty Dont Send That Event)
+
+                if (!string.IsNullOrEmpty(vehicle.pilotId)) // OnPilotMode Set That Value if Value exist have Remote Player Inside (Need Detach Before Destroy gameObject)
+                {
+                    PlayerManager remotePlayer = NitroxServiceLocator.LocateService<PlayerManager>();
+                    Optional<RemotePlayer> player = remotePlayer.Find(vehicle.pilotId);
+                    if (player.IsPresent())
+                    {
+                        RemotePlayer RP = player.Get();
+                        RP.SetVehicle(null);
+                        RP.SetSubRoot(null);
+                        RP.SetPilotingChair(null);
+                        RP.AnimationController.UpdatePlayerAnimations = true;
+                    }
+                }
+            }
+        }
+
+        public void OnPilotMode(Vehicle vehicle, byte type) // Set & Unset Remote Player Value Inside of Vehicle Sender
+        {
+            if (!string.IsNullOrEmpty(vehicle.gameObject.GetGuid()))
+            {
+                VehicleOnPilotMode mode = new VehicleOnPilotMode(vehicle.gameObject.GetGuid(), GuidHelper.GetGuid(Player.main.gameObject), type);
+                packetSender.Send(mode);
+                Log.Info("Sender: " + mode.ToString());
+            }
+            
+        }
+
+        public void OnPilotModeSet(string vehicleGuid, string playerGuid, byte type) // Set & Unset Remote Player Value Inside of Vehicle Receiver
+        {
+            Log.Info("Receiver: VehicleGuid: " + vehicleGuid + " PlayerGuid: " + playerGuid + " Type: " + type);
+
+            Optional<GameObject> Vehicle = GuidHelper.GetObjectFrom(vehicleGuid);
+
+            if (Vehicle.IsPresent())
+            {
+                Log.Info("Object Found");
+                GameObject Go = Vehicle.Get();
+                Vehicle V = Go.GetComponent<Vehicle>();
+
+                if (V != null)
+                {
+                    if (type == 0)
```

Comment:
what does type of 0 mean?  Can you add a comment?

Reply:
I'm against comments if usage can be made clear with the correct type and name (enum, integer constant, or even a bool should be sufficient in this case).